edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,37 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let (lock, limit) = if let ConcurrencyThrottling::Limited(limit) = concurrency {
+            (Some(Arc::new(AtomicU32::new(*limit))), *limit)
+        } else {
+            (None, 0 as u32)
+        };
+
         let srv = incoming.for_each(move |(socket, addr)| {
-            let protocol = protocol.clone();
 
+            if let Some(lock) = lock.clone() {
+                let limit_reached = lock.fetch_update( Ordering::AcqRel,
+                    Ordering::Acquire,
+                    |current| {
+                        if current == 0 {
+                            None
+                        } else {
+                            Some(current - 1)
+                        }},
+                    ).is_err();
+
+                if limit_reached {
+                    {
```

Comment:
nit: these extra braces are likely unnecessary as limit is u32 I think which is copy by default, so it shouldn't cause the borrow checker to slap your face :)

Reply:
`u32` is `Copy`, but also `error!` borrows its args, but also regardless of either of those things these braces don't do anything.

Reply:
Fixed

Reply:
Hasn't been fixed.

Reply:
Strangely, it was added by cargo fmt. I never put them in the first place... They got added back automatically after I saved...
I restarted the VS remote, seems fine now.